[sherwynNg] iP

SherwynNg opened the pull request:
### DukePro
> "Your mind is for having ideas, not holding them." -David Allen [(source)](https://dansilvestre.com/productivity-quotes/)

DukePro frees your mind of having to remember things you need to do. It's
  - text-based
  - easy to learn
  - ~~FAST~~ _SUPER FAST_ to use

All you need to do is,
  1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
  2. double-click it.
  3. add your tasks.
  4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)
---
If you are a Java programmer, you can use it to practice Java too. Here's the <span style="color:red">`main`</span> method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
````

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Commit:
added "Level 0. Rename, Greet, Exit"

Commit:
added "level-1, echo"

Commit:
removed printwriter

Commit:
added "Level 2. Add, list"

Commit:
added Task class

Commit:
added "Level 3. Mark as Done"

Commit:
added "Deadline class"

Commit:
added "Event class"

Commit:
added "Todo class"

Commit:
added "Level 4. ToDos, Events, Deadlines"

Commit:
Revert "added "Level 4. ToDos, Events, Deadlines""

This reverts commit db27ae4645611269218fb7de7d9ebc966658b03f.

Commit:
added "toString method"

Commit:
added "Level 4. ToDos, Events, Deadlines"

Commit:
Changed a typo

Commit:
Added "Input and Expected file for testing"

Commit:
Added "DukeExceptions class"

Commit:
Included different test cases for error handling

Commit:
Added "Level 5. Handle Errors"

Commit:
added exceptions to check for empty list and delete action

Commit:
updated input and expected files to include new test cases

Commit:
added "Level 6. Delete"

Commit:
add Enums testcase

Commit:
add enums

Commit:
handle corrupted file

Commit:
add Level-7: Save

Commit:
add text file

Commit:
add Level 8. Dates and Times

Commit:
Merge branch 'branch-level-8'

* branch-level-8:
  add Level 8. Dates and Times

Commit:
add new exception

Commit:
Merge branch 'branch-level-8'

* branch-level-8:
  add new exception

Commit:
add more OOP classes, Ui, Storage, Parser and TaskList

Commit:
Merge branch 'branch-A-MoreOOP'

* branch-A-MoreOOP:
  add more OOP classes, Ui, Storage, Parser and TaskList

Commit:
add packages

Commit:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  add packages

Commit:
Merge branch 'add-gradle-support'

* add-gradle-support:
  Bump gradle and lib version
  Add Gradle support
  docs/README.md: Tweak document template

Commit:
add test cases to test deadline task toString() and adding of tasks

Commit:
change shadowJar in build.gradle

Commit:
add Javadocs for the different methods

Commit:
add Level 9. Find

Commit:
add checkstyle to gradle

Commit:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  add Javadocs for the different methods

Commit:
Merge branch 'branch-Level-9'

* branch-Level-9:
  add Level 9. Find

Commit:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  add checkstyle to gradle

Commit:
fix some indentations

Commit:
change file path for gradle

Commit:
create folder if folder not found

Commit:
change indentation for switch cases

Commit:
add checkstyle and edit code

Commit:
Merge branch 'branch-A-checkstyle'

* branch-A-checkstyle:
  add checkstyle and edit code

justincred commented on src/main/java/academicweapon/Duke.java:
```diff
@@ -0,0 +1,146 @@
+ * Duke is a task management application that allows users to interact with tasks through a command-line interface.
+ * It supports various command for managing tasks such as adding, listing, marking and deleting tasks.
+ */
+public class Duke {
```
Maybe you can rename your chatbot name to another name, same with DukeException

justincred commented on src/main/java/academicweapon/Duke.java:
```diff
@@ -0,0 +1,146 @@
+
+            switch (action) {
+            case FIND:
+                ArrayList<String> strLst = this.tasks.findKeyword(actions.get(1));
```
I think it would be nice to name strLst into something more direct like what the findKeyword function returns

cocoanautz replied:
Collection names should also be in plural

justincred commented on src/main/java/academicweapon/task/Event.java:
```diff
@@ -0,0 +1,48 @@
+ */
+public class Event extends Task {
+
+    protected String from;
```
Maybe you can be more specific such as fromTime and i think it needs to be stored as localdate or localdatetime format instead, same as your deadline class
![image](https://github.com/nus-cs2103-AY2324S2/ip/assets/95204474/576f60b8-3db3-45e4-bc81-f5b94dd2fd11)
It would also be better to be able to implement certain localdate features such as sorting based on time

justincred commented on src/main/java/academicweapon/task/TaskList.java:
```diff
@@ -0,0 +1,157 @@
+    }
+
+    /**
+     * Consturcts a TaskList based on the provided list of string representations of tasks
```
aha a spelling mistake here, should be Constructs

justincred reviewed (commented):
overall nicely done, only some very minor coding standard issues!

Commit:
finish javaFX tutorial 1

Commit:
finished tutorial 2 for javafx

Commit:
finish javaFx tutorial 3

cocoanautz commented on src/main/java/academicweapon/Duke.java:
```diff
@@ -0,0 +1,146 @@
+                    }
+                }
+                break;
+            case LIST:
```
Noticed that your switch case follows the coding standard! Good job.

cocoanautz commented on src/main/java/academicweapon/storage/FileManager.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * File route for storing the task list data file.
+     */
+    private static final String fileRoute = "./src/main/data/acadList.txt";
```
Constant names should be in screaming_snake_case

cocoanautz commented on src/main/java/academicweapon/storage/FileManager.java:
```diff
@@ -0,0 +1,145 @@
+package academicweapon.storage;
+
+import academicweapon.exceptions.DukeExceptions;
+import academicweapon.task.Deadline;
+import academicweapon.task.Event;
+import academicweapon.task.Task;
+import academicweapon.task.Todo;
```
Great job in making sure that your import statements are orderly.

cocoanautz commented on src/main/java/academicweapon/ui/Ui.java:
```diff
@@ -0,0 +1,113 @@
+     * Displays a horizontal line as a separator in the console.
+     */
+    public static void showLine() {
+        System.out.println("____________________________________________________________");
```
You can probably store this as a constant, which would make sure all lines are the same length.

cocoanautz reviewed (commented):
LGTM. Just some minor nits to deal with, mostly related to variable names.

Commit:
finish level 10, GUI

Commit:
Merge branch 'branch-level-10'

* branch-level-10:
  finish level 10, GUI
  finish javaFx tutorial 3
  finished tutorial 2 for javafx
  finish javaFX tutorial 1

Commit:
build a new Jar file

Commit:
showLoadingError() method is printing an error message.

The error message is fixed and should be consistent.

A named constant, LOADING_ERROR, is created.

It makes the code easier to understand because the name tells us more about the meaning of the number.

Commit:
showBye() method is printing an exit message.

The exit message is fixed and should be consistent.

A named constant, BYE_MESSAGE, is created.

It makes the code easier to understand because the name tells us more about the meaning of the number.

Commit:
add assertions to define assumptions about the program state so that the runtime can verify them

Commit:
improve code quality by using guard clauses

SherwynNg committed:
Merge pull request #1 from SherwynNg/branch-A-Assertions

add assertions to define assumptions about the program state so that …

Commit:
Merge branch 'master' of https://github.com/SherwynNg/ip

* 'master' of https://github.com/SherwynNg/ip:
  add assertions to define assumptions about the program state so that the runtime can verify them

SherwynNg committed:
Merge pull request #2 from SherwynNg/branch-A-CodeQuality

improve code quality by using guard clauses

Commit:
Merge branch 'master' of https://github.com/SherwynNg/ip

* 'master' of https://github.com/SherwynNg/ip:
  improve code quality by using guard clauses

Commit:
add B-Reminders

Commit:
Merge branch 'branch-B-Reminders'

* branch-B-Reminders:
  add B-Reminders

Commit:
remove extra calling of Duke

Commit:
changed Duke to AcademicWeapon

Commit:
added title

Commit:
edit dialogbox to show full message

Commit:
add exit after command bye

Commit:
updated Readme
